### PR TITLE
Spec for staking module differences

### PR DESCRIPTION
**Pull request by S1nus**

### CODE REVIEW

File: specs/src/specs/staking_module.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Staking (Differences from Tendermint)
+
+## Abstract
+
+Rollkit will support customizable functionality for verifying the correctness of a block proposer. However, it currently works by delegating this functionality to the Application, by checking that the `AggregatorsHash` of a header corresponds to the `NextAggregatorsHash` of the previous, where setting the `NextAggregtorsHash` is the responsibility of the ABCI Application.
+
+## Assumptions and Considerations
```
[the remark this answers is not in the document]
Comment (S1nus) [reply specs/src/specs/staking_module.md:7]: "tendermint staking" does not exist. the ABCI App simply sets the validator set

[the remark this answers is not in the document]
Comment (nashqueue) [reply specs/src/specs/staking_module.md:7]: Yes, we use the staking module for that.

**Review by nashqueue — CHANGES_REQUESTED** [review]
This spec aims to check how we use the staking module vs. how it is used in tendermint. Please elaborate on that.

**Comment by S1nus** [comment]
> This spec aims to check how we use the staking module vs. how it is used in tendermint. Please elaborate on that.

we use it in exactly the same way?

**Comment by nashqueue** [comment]
> > This spec aims to check how we use the staking module vs. how it is used in tendermint. Please elaborate on that.
> 
> don't we use it in exactly the same way?

The point of this spec is to exactly figure this out and to make sure to capture the nuances if there are differences.

**Comment by S1nus** [comment]
As of https://github.com/rollkit/rollkit/issues/1291, we decided that Rollkit vA will control the proposer entirely, ignoring all validator set updates from the ABCI app.

This is being implemented in the `Header.Verify(...)` method, where it enforces that the `ProposerAddress` must be equal to the `ProposerAddress` of the previous header. If implemented properly, this ensures the entire chain is signed by the same person (the centralized sequencer).

**Pull request closed without merging**